hnn_core/cell.py

```diff
@@ -293,18 +306,28 @@ def move_to_pos(self):
     # 2. a list needs to be created with a Dipole (Point Process) in each
     #    section at position 1
     # In Cell() and not Pyr() for future possibilities
-    def insert_dipole(self, yscale):
+    def insert_dipole(self, p_secs, dipole_sec):
         """Insert dipole into each section of this cell.
 
         Parameters
         ----------
-        yscale : dict
-            Dictionary of length scales to calculate dipole without
-            3d shape.
+        p_secs : nested dict
+            Dictionary with keys as section name.
+            p_secs[sec_name] is a dictionary with keys
+            L, diam, Ra, cm, syns and mech.
+            syns is a list specifying the synapses at that section.
+            The properties of syn are specified in p_syn.
+            mech is a dict with keys as the mechanism names. The
+            values are dictionaries with properties of the mechanism.
+        dipole_sec : str
+            The name of the section along which dipole is to
+            be computed.
         """
         self.dpl_vec = h.Vector(1)
         self.dpl_ref = self.dpl_vec._ref_x[0]
 
+        yscale = _get_yscale(p_secs, 'apical_trunk')
```

Comment:
Re: previous comment, `_get_yscale()` could return the y-scale for a single section called within the for loop below at L334.

Reply:
I'm confused about this too! I think variable & function naming should be better. "yscale" is actually a projection of each (linear) section onto the line defined by the orientation of the apical dendrite. How about:

```python
apical_projs = _get_sect_projs_onto_apical_orientation(p_secs, orientation='apical_trunk')
```